src/main/java/com/formulasearchengine/mathmlquerygenerator/NtcirTopicReader.java

```diff
@@ -0,0 +1,110 @@
+package com.formulasearchengine.mathmlquerygenerator;
+
+import com.formulasearchengine.mathmltools.xmlhelper.NonWhitespaceNodeList;
+import com.formulasearchengine.mathmltools.xmlhelper.XMLHelper;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Moritz on 08.11.2014.
+ * <p/>
+ * Reads the topic format specified in
+ * http://ntcir-math.nii.ac.jp/wp-content/blogs.dir/13/files/2014/05/NTCIR11-Math-topics.pdf
+ */
+public class NtcirTopicReader {
+    public static final String NS_NII = "http://ntcir-math.nii.ac.jp/";
+    private final Document topics;
+    private final List<NtcirPattern> patterns = new ArrayList<>();
+    private final QVarXQueryGenerator queryGenerator;
+
+    public NtcirTopicReader(Document topics) {
+        this.topics = topics;
+        queryGenerator = new QVarXQueryGenerator();
+    }
+
+    public NtcirTopicReader(File topicFile) throws ParserConfigurationException, IOException, SAXException {
+        DocumentBuilder documentBuilder = XMLHelper.getDocumentBuilder(true);
+        topics = documentBuilder.parse(topicFile);
+
+        //TODO: Find out how this code duplication can be avoided in Java.
+        queryGenerator = new QVarXQueryGenerator();
```

Comment:
Instead of 39 - 43 write:
this( XMLHelper.getDocumentBuilder(true).parse(topicFile) );